OpenRA.Mods.Common/UtilityCommands/ExtractEmmyLuaAPI.cs

```diff
@@ -0,0 +1,358 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using OpenRA.Scripting;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.UtilityCommands
+{
+	// See https://emmylua.github.io/annotation.html for reference
+	class ExtractEmmyLuaAPI : IUtilityCommand
+	{
+		string IUtilityCommand.Name => "--emmy-lua-api";
+
+		bool IUtilityCommand.ValidateArguments(string[] args)
+		{
+			return true;
+		}
+
+		[Desc("Generate EmmyLua API annotations for use in IDEs.")]
+		void IUtilityCommand.Run(Utility utility, string[] args)
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			Game.ModData = utility.ModData;
+
+			var version = Game.ModData.Manifest.Metadata.Version;
```

Comment:
This code uses `Game.ModData` even though [line 34](https://github.com/OpenRA/OpenRA/pull/19763/commits/bdd2c820a69322828eb3a210008178fb688090d7#diff-135da93642d39ce276e6d7dbc3b4caa8bdb476af982c44ba7a827642f35d547aR34) refers to setting `Game.ModData` as a hack.
One solution:
```suggestion
			var version = utility.ModData.Manifest.Metadata.Version;
```